### ISSUE TITLE
I can't manage to get all Artworks to upload with 2.0.2 version of foo_discord_rich

### ISSUE THREAD

**Issue by DaRealMiku** [TheQwertiest/foo_discord_rich#91]
As the title says, I always used the 1.3.0-beta of Discord Rich Presence Integration, and using python has worked flawless, however, I decided to update yesterday to the 2.0.2 using python aswell, but it seems most artwork covers won't upload to Discord and show default foobar image. 

This is with 1.3.0
![image](https://github.com/user-attachments/assets/4051d2c2-b7a3-4759-8b6c-c4261f61ac28)
Command used
![image](https://github.com/user-attachments/assets/d112420a-f285-45fc-9937-51d26d0e03d8)
and then C:\Path\To\script.py (script? i guess?)

but when using the 2.0.2
![image](https://github.com/user-attachments/assets/f127e7a8-4675-499a-a4fd-2fcf6d0f3fd4)
Command used:
![image](https://github.com/user-attachments/assets/df1fafce-0759-4958-8a69-03920d4ac1c3)
(which is the same, copypasted)

Is this an issue at my end? Thank you for your attention
I'm also using foobar 32 bit.

I don't know if I need any other info to be dropped, thank you.

**Comment by UNITEMystic** [comment]
i am having the same issue, except i am using the standalone binary for imgur. There is also no context menu options for working with this update unlike how there were context menu options to upload artwork/delete artwork/etc in 1.3.0 beta. Ive managed to get some albums to attempt to upload but either the process times out very fast or i have to go through a cycle of manipulating the art cache.

**Comment by rdignarz** [comment]
I'd like to use this feature also but I never had a python script to make this work. My artworks are all embedded. Is there any guidance on what is needed? Or is there a script already uploaded here that I can't seem to find?

**Comment by PlaylistsTrance** [comment]
I use a Python script that uploads to catbox.moe, modified a bit to support saving errors to a log file and automatically resize large images to 100x100.

The command I use is:

`"C:\path\to\python.exe" "C:\path\to\foobar_cover_upload.py"`

```python
from io import BytesIO
import logging
import os
import sys
import traceback

from PIL import Image
import requests

logging.basicConfig(filename="foobar_cover_upload.log", encoding="utf-8",
                    format="%(asctime)s %(levelname)-8s %(message)s")


try:
    sys.stdin.reconfigure(encoding='utf-8')  # noqa
    filename = sys.stdin.read().strip()

    api_url = "https://catbox.moe/user/api.php"
    data = {
        'reqtype': (None, 'fileupload'),
    }

    if os.path.getsize(filename) > 262144:  # .25 MB in bytes
        img = Image.open(filename)
        img.thumbnail((100, 100), Image.LANCZOS, reducing_gap=3.0)
        with BytesIO() as buffer:
            if img.mode == "RGBA":
                background = Image.new("RGBA", img.size, (255, 255, 255))
                alpha_composite = Image.alpha_composite(background, img).convert("RGB")
                alpha_composite.save(buffer, "jpeg", quality=80)
            else:
                img.save(buffer, format="jpeg", quality=80)
            buffer.seek(0)
            buffer.name = "cover.jpg"
            data['fileToUpload'] = buffer
            r = requests.post(api_url, files=data)
    else:  # Image does not need to be resized
        with open(filename, 'rb') as f:
            data['fileToUpload'] = f
            r = requests.post(api_url, files=data)

    if not r.ok:
        logging.error(r.text)
        print(r.text[:1000])
        sys.exit(1)

    if not r.text:
        logging.error(f"Received empty string instead of URL for {filename}")
        sys.exit(1)

    print(r.text, end='')

except Exception as e:  # noqa
    logging.error(str(e))
    traceback.print_exc(file=sys.stdout)
    sys.exit(1)
```

Note that this does require modules requests and Pillow (`pip install requests Pillow`).

**Comment by DaRealMiku** [comment]
> I use a Python script that uploads to catbox.moe, modified a bit to support saving errors to a log file and automatically resize large images to 100x100.
> 
> The command I use is:
> 
> `"C:\path\to\python.exe" "C:\path\to\foobar_cover_upload.py"`
> 
> ```python
> from io import BytesIO
> import logging
> import os
> import sys
> import traceback
> 
> from PIL import Image
> import requests
> 
> logging.basicConfig(filename="foobar_cover_upload.log", encoding="utf-8",
>                     format="%(asctime)s %(levelname)-8s %(message)s")
> 
> 
> try:
>     sys.stdin.reconfigure(encoding='utf-8')  # noqa
>     filename = sys.stdin.read().strip()
> 
>     api_url = "https://catbox.moe/user/api.php"
>     data = {
>         'reqtype': (None, 'fileupload'),
>     }
> 
>     if os.path.getsize(filename) > 262144:  # .25 MB in bytes
>         img = Image.open(filename)
>         img.thumbnail((100, 100), Image.LANCZOS, reducing_gap=3.0)
>         with BytesIO() as buffer:
>             if img.mode == "RGBA":
>                 background = Image.new("RGBA", img.size, (255, 255, 255))
>                 alpha_composite = Image.alpha_composite(background, img).convert("RGB")
>                 alpha_composite.save(buffer, "jpeg", quality=80)
>             else:
>                 img.save(buffer, format="jpeg", quality=80)
>             buffer.seek(0)
>             buffer.name = "cover.jpg"
>             data['fileToUpload'] = buffer
>             r = requests.post(api_url, files=data)
>     else:  # Image does not need to be resized
>         with open(filename, 'rb') as f:
>             data['fileToUpload'] = f
>             r = requests.post(api_url, files=data)
> 
>     if not r.ok:
>         logging.error(r.text)
>         print(r.text[:1000])
>         sys.exit(1)
> 
>     if not r.text:
>         logging.error(f"Received empty string instead of URL for {filename}")
>         sys.exit(1)
> 
>     print(r.text, end='')
> 
> except Exception as e:  # noqa
>     logging.error(str(e))
>     traceback.print_exc(file=sys.stdout)
>     sys.exit(1)
> ```
> 
> Note that this does require modules requests and Pillow (`pip install requests Pillow`).

Does this requiere a created api on my own? or?

**Comment by PlaylistsTrance** [comment]
Nope, just what I posted. Actually, you don't even need to find the whole path of your Python install, the command could just be `py C:\path\to\foobar_cover_upload.py`, after having installed requests and Pillow with `py -m pip install requests Pillow`.
